src/diff/amfGraphDifferencer.ts

```diff
@@ -171,6 +171,15 @@ function parseGraphDelta(
   return graphChanges;
 }
 
+/**
+ * Returns true if type of amf graph node is defined
+ * @param nodeType - Types of amf graph node
+ */
+function isNodeTypeDefined(nodeType: string[]): boolean {
+  //source map nodes may not have type defined
+  return nodeType != null && nodeType.length > 0;
+}
```

Comment:
```suggestion
function isNodeTypeDefined(nodeType: string[]): boolean {
  //source map nodes may not have type defined
  return nodeType.length > 0;
}
```
If `nodeType` is always `string[]`, then this is sufficient.

```suggestion
function isNodeTypeDefined(nodeType: string[] | undefined | null): nodeType is string[] {
  //source map nodes may not have type defined
  return nodeType?.length > 0;
}
```
If `nodeType` is not always `string[]`, then (1) the type should reflect that, (2) the return value should be a type predicate so typescript can use the information, and (3) you can use the safe navigation operator to be more concise.